packages/integrations/solid/src/dependencies.ts

```diff
@@ -0,0 +1,56 @@
+// This file is a fork of vite-plugin-solid.
+// Original: https://github.com/solidjs/vite-plugin-solid/blob/03130c8a0a2ceaab9a07e16f1e1df832b996e1b8/src/index.ts#L251-L297
+// License: MIT (https://github.com/solidjs/vite-plugin-solid/blob/03130c8a0a2ceaab9a07e16f1e1df832b996e1b8/package.json#L38)
+
+import fs from 'fs';
+import path from 'path';
+import { createRequire } from 'module';
+import { fileURLToPath } from 'url';
+
+function containsSolidField(fields: Record<string, any>) {
+	const keys = Object.keys(fields);
+	for (let i = 0; i < keys.length; i++) {
+		const key = keys[i];
+		if (key === 'solid') return true;
+		if (typeof fields[key] === 'object' && containsSolidField(fields[key])) return true;
+	}
+	return false;
+}
+
+export function getSolidDeps(root: URL) {
+	const pkgPath = path.join(fileURLToPath(root), 'package.json');
+	if (!fs.existsSync(pkgPath)) {
+		console.log('No package.json found at project root');
```

Comment:
need need the eslint disable comment here

Reply:
Ah yeah I just copy-pasted this, will clean up the lints